app/scripts/views/mixins/signin-mixin.js

```diff
@@ -145,21 +147,22 @@ define(function (require, exports, module) {
         .then(this.navigate.bind(this, this.model.get('redirectTo') || 'settings', {}, navigateData));
     },
 
-    _engageSignInForm (event) {
-      if (event && event.target) {
-        var target = this.$el.find(event.target).attr('id');
-
-        if (target === 'have-account') {
-          // if the user clicks on 'have-account' we count that as flow event instead of the 'engage' event.
-          // Details: https://github.com/mozilla/fxa-content-server/pull/4221
-          this.logEvent('flow.have-account');
-          return;
-        }
+    _clickLink (event) {
+      if (event && event.target && event.target.id) {
+        // Each link on the sign-in form emits a different event,
+        // named according to its id.
+        this.logFlowEvent(event.target.id, this.viewName);
       }
+    },
 
-      // user has engaged with the sign in, sign up or force auth form
-      // the flow event will be different depending on the view name
-      this.logEventOnce(`flow.${this.viewName}.engage`);
-    }
+    _engageSignInForm () {
```

Comment:
Both `_engageSignInForm` and `_submitSignInForm` will be called on the signup form, is that alright? This is where the signin-mixin is a stretch for a few functions, do `_clickLink`, `_engageSignInForm`, and `submitSignInForm` make more sense on the [flow-begin-mixin](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/views/mixins/flow-begin-mixin.js) or even in [form.js](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/views/form.js)?